definitions/npm/async_v1.x.x/flow_>=v0.28.x/async_v1.x.x.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+declare module 'async' {
+  declare type Collection<TItem> = Iterable<TItem> | { [key: string]: TItem };
+
+  declare type IterateeCallback<TError, TResult> = (err?: TError, result?: TResult) => void;
+  declare type Callback<TError, TResult> = (err?: TError, result?: TResult) => mixed;
+
+  // Collections
+  declare function each<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (item: TItem, callback: IterateeCallback<TError, void>) => mixed,
+    callback?: Callback<TError, void>
+  ): void;
+  declare function forEachOf<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (value: TItem, key: string | number, callback: IterateeCallback<TError, void>) => mixed,
```

Comment:
An alternative might be to write it like this:

```js
type $Iterable<TKey: number, TValue> = Iterable<TValue>;
type Collection<TKey, TValue> = $Iterable<TKey, TValue> | { [key: TKey]: TValue };
```

[But that doesn't work either](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgAkAkhngE4CGARjHgDwAqA0nlgFxgB2ArgLbUKAGjCMAapRg88APjABeMKQo06TCVNkBuTLgIBhODDoBjDAEs4XJqywjxk6XMUkyVWgxZt7Gp2AA+YADeYADaANZsnF5YALrRvgQAvjq6+EpulGR4+pIw1JQm4UwAouTkcOT2AEp4AM48MBjOYAAUFOQA-NFlFVVg5PWNGN2itQ1NAJQKcgBucOYAJjrY6bnGBUWl5ZU1Q00t7eWjjL27A-sj0ePD0-JyfOYAHnjL6FA8XGaWXGBQlSVCgALADyUBs3lEiXsZ3IMlaqDAYBMRlMFisnEMxjw3ysELsUMcsiEiLA5ky2U4rVmRISRJEkQ4olsIhMeU24U4yio2XW+UKxVOO368yWMjuD2erxJSLZGwFoz5HO2fREosWMlQk046uCSXeAOBYNaIUonAA5ABBc1gJIian0sCMzi8ATCZHsgUSvWTHT-ciAkygqAmsBmsBWm12to0zQMqJgOoYcjmLgAc1ZnqK3qCSV9qCAA).